contrib/src/main/java/com/datatorrent/contrib/parser/LogSchemaDetails.java

```diff
@@ -0,0 +1,230 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.contrib.parser;
+
+import org.codehaus.jettison.json.JSONArray;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * <p>
+ * This is schema that defines fields and their regex
+ * The operators use this information to validate the incoming tuples.
+ * Information from JSON schema is saved in this object and is used by the
+ * operators
+ * <p>
+ * <br>
+ * <br>
+ * Example schema <br>
+ * <br>
+ * {@code{ "fields": [{"field": "host","regex": "^([0-9.]+)"},
+ * {"field": "userName","regex": "(.*?)"},
+ * {"field": "request","regex": "\"((?:[^\"]|\")+)\""},
+ * {"field": "statusCode","regex": "(\\d{3})"},
+ * {"field": "bytes","regex": "(\\d+|-)"}]}
+ */
+public class LogSchemaDetails
+{
+  /**
+   * This holds the list of field names in the same order as in the schema
+   */
+  private List<String> fieldNames = new LinkedList();
+
+  private List<Field> fields = new LinkedList();
+
+  /**
+   * This holds regex pattern for the schema
+   */
+  private String pattern;
+
+  public LogSchemaDetails(String json)
+  {
+    try {
+      initialize(json);
+      createPattern();
+    } catch (JSONException | IOException e) {
+      logger.error("{}", e);
+      throw new IllegalArgumentException(e);
+    }
+  }
+
+  /**
+   * For a given json string, this method sets the field members
+   * @param json
+   * @throws JSONException
+   * @throws IOException
+   */
+  private void initialize(String json) throws JSONException, IOException
+  {
+    JSONObject jsonObject = new JSONObject(json);
+    JSONArray fieldArray = jsonObject.getJSONArray("fields");
+
+    for(int i = 0; i < fieldArray.length(); i++) {
+      JSONObject obj = fieldArray.getJSONObject(i);
+      Field field = new Field(obj.getString("field"), obj.getString("regex"));
+      this.fields.add(field);
+      this.fieldNames.add(field.name);
+    }
+  }
+
+  /**
+   * creates regex group pattern from the regex given for each field
+   */
+  public void createPattern()
+  {
+    StringBuffer pattern = new StringBuffer();
+    for(Field field: this.getFields()) {
+      pattern.append(field.getRegex()).append(" ");
+    }
+    logger.info("Created pattern for parsing the log {}", pattern.toString().trim());
+    this.setPattern(pattern.toString().trim());
```

Comment:
If we're trimming the final result, shouldn't we also trim each field inside the loop ?